src/coreclr/jit/compiler.hpp

```diff
@@ -116,7 +116,7 @@ inline bool genMaxOneBit(T value)
 template <typename T>
 inline bool genExactlyOneBit(T value)
 {
-    return ((value != 0) && genMaxOneBit(value));
+    return BitOperations::PopCount(value) == 1;
```

Comment:
Given we need a branch to test for `popcnt` support, I expect the old logic may actually be faster as it generates:
```
    test edi, edi
    jz false
    lea eax, [rdi - 1]
    test edi, eax
    sete al
    ret
false:
    xor eax, eax
    ret
```

Reply:
So existing logic has 2 branches vs. just 1 branch with `popcount()`. Do you think existing logic would still be faster?

Reply:
The existing logic is just 1 branch. `setcc` is considered branchless and is specially handled by the CPU. The `lea eax, [rdi - 1]`, `test edi, eax`, `sete al` is 3 cycles which is the same as for `popcnt`.

So it'd likely balance out, but with there now being 2 branches for anyone with "very old" hardware. I don't have a particular preference for which we do given that `popcnt` has been around for 15 years now, so we're unlikely to have people without it.

Reply:
> So it'd likely balance out

In that case, I don't think I should go ahead of using `popcnt` in `genExactlyOneBit()`, and just replace the existing bit twiddling logic with `popcnt` on supported hardware. That way the future `BitOperations::PopCount()` consumer will be optimized from `popcnt`. Thoughts?

Reply:
That seems reasonable to me.